cms/djangoapps/contentstore/rest_api/v0/views/details_settings.py

```diff
@@ -0,0 +1,39 @@
+""" API Views for course details settings """
+
+from opaque_keys.edx.keys import CourseKey
+from rest_framework.request import Request
+from rest_framework.views import APIView
+from xmodule.modulestore.django import modulestore
+
+from cms.djangoapps.models.settings.encoder import CourseSettingsEncoder
+from common.djangoapps.student.auth import has_studio_read_access, has_studio_write_access
+from common.djangoapps.util.json_request import JsonResponse, expect_json
+from openedx.core.lib.api.view_utils import DeveloperErrorViewMixin, verify_course_exists, view_auth_classes
+from openedx.core.djangoapps.models.course_details import CourseDetails
+
+from ....views.course import update_course_details_settings
+
+
+@view_auth_classes(is_authenticated=True)
+@expect_json
+class CourseDetailsSettingsView(DeveloperErrorViewMixin, APIView):
+
+    @verify_course_exists()
+    def get(self, request: Request, course_id: str):
+        course_key = CourseKey.from_string(course_id)
+        if not has_studio_read_access(request.user, course_key):
+            self.permission_denied(request)
+        course_details = CourseDetails.fetch(course_key)
+        return JsonResponse(
+            course_details,
+            # encoder serializes dates, old locations, and instances
+            encoder=CourseSettingsEncoder
+        )
+
+    @verify_course_exists()
+    def patch(self, request: Request, course_id: str):
+        course_key = CourseKey.from_string(course_id)
+        if not has_studio_write_access(request.user, course_key):
+            self.permission_denied(request)
+        course_block = modulestore().get_course(course_key)
+        return update_course_details_settings(course_key, course_block, request)
```

Comment:
Do these endpoints generate meaningful Swagger docs? If not, can we add some declarations / docs, like for [advanced settings](https://github.com/openedx/edx-platform/blob/518bbe8d636463e91fe0aa0aed10fb80a5d0af3d/cms/djangoapps/contentstore/rest_api/v0/views/advanced_settings.py#L120-L179)?

Reply:
Yeah it will require the `@apidocs.schema` decorator. I was planning on adding it once we agree on the implementation. Will do so now.